### DISCUSSION TITLE
error: path must be absolute or specify root to res.sendFile

### DISCUSSION THREAD

**Discussion by Hagbrain** [requarks/wiki#5223] [Error / Bug Report]
Unable to view images - res.SendFile
Unable to view images after update. Console shows an error and the page just shows the filename instead of an image.

`[MASTER] error: path must be absolute or specify root to res.sendFile`

Theres a solution
-----------------------------
Suggested solution:

this line:

await sendFile(cachePath, { dotfiles: 'deny', root: "/opt/wiki/" })

replace with this line:

try {
await sendFile(cachePath, { dotfiles: 'deny' })
}catch (error) {
await sendFile(cachePath, { dotfiles: 'deny', root: '/opt/wiki/' })
}
----------------------------

**But what file do i need to edit?**

Runing lastest wikijs dockerized

Kind Regards
Dom

**Comment by rhod92** [comment]
I think the file is "server/models/assets.js" inside your wiki

**Reply by rhod92** [reply]
I replaced this part of the assets.js code with:

```
static async getAssetFromCache(assetPath, cachePath, res) {
  try {
    await fs.access(cachePath, fs.constants.R_OK)
  } catch (err) {
    return false
  }

  const sendFile = Promise.promisify(res.sendFile, { context: res })
  res.type(path.extname(assetPath))

  try {
    await sendFile(cachePath, { dotfiles: 'deny' })
  } catch (error) {
    await sendFile(cachePath, { dotfiles: 'deny', root: '/opt/wiki/' })
  }

  return true
}
```
